### PR TITLE
Add metdata resource to AD role + ENV Var adjustments

### PR DESCRIPTION
**Pull request by freecaykes**
## Description

New RBAC Permissions for AD Pods querying AD API and Updates in ENV_VARs 

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: pkg/render/intrusion_detection.go [hunk 1]
```diff
@@ -1504,20 +1518,24 @@ func (c *intrusionDetectionComponent) getBaseADDetectorsPodTemplate(podTemplateN
 						},
 						Env: []corev1.EnvVar{
 							{
-								Name: "ELASTIC_HOST",
+								Name: "ES_HOST",
```
Comment (rene-dekker) [inline pkg/render/intrusion_detection.go:1521]: I think the preferred way would be to actually use the existing function `relasticsearch.ContainerDecorate(...)`. For an example, see compliance.go. We have a convenient function that normalizes all of these fields. Would it be possible to modify the adDetectorsImage, such that these standard fields can be continued to be supported?

File: pkg/render/intrusion_detection.go [hunk 2]
```diff
@@ -1537,10 +1555,6 @@ func (c *intrusionDetectionComponent) getBaseADDetectorsPodTemplate(podTemplateN
 								Name:      "MODEL_STORAGE_API_TOKEN",
 								ValueFrom: secret.GetEnvVarSource(adDetectorServiceAccountName, "token", false),
 							},
-							{
```
Comment (rene-dekker) [inline pkg/render/intrusion_detection.go:1540]: Your pod and container should be using:
```
c.cfg.TrustedCertBundle.Volume(),
c.cfg.TrustedCertBundle.VolumeMount(...),
```
for mounting the certificate bundle, which includes already the `esgwCertificate` that allows you to trust elasticsearch.

File: pkg/render/intrusion_detection.go (hunk 1 above)
Comment (rene-dekker) [reply pkg/render/intrusion_detection.go:1521]: And this:
```
					{
						Name: "es-certs",
						VolumeSource: corev1.VolumeSource{
							Secret: &corev1.SecretVolumeSource{
								SecretName: relasticsearch.PublicCertSecret,
								Items: []corev1.KeyToPath{
									{Key: "tls.crt", Path: "es-ca.pem"},
								},
							},
						},
					},
```
Can then be removed.

File: pkg/render/intrusion_detection.go (hunk 2 above)
Comment (rene-dekker) [reply pkg/render/intrusion_detection.go:1540]: And this:
```
					{
						Name: "es-certs",
						VolumeSource: corev1.VolumeSource{
							Secret: &corev1.SecretVolumeSource{
								SecretName: relasticsearch.PublicCertSecret,
								Items: []corev1.KeyToPath{
									{Key: "tls.crt", Path: "es-ca.pem"},
								},
							},
						},
					},
```
Can then be removed.

Comment (freecaykes) [reply pkg/render/intrusion_detection.go:1540]: https://github.com/tigera/operator/pull/2074/commits/72b90570405d14a3140de0aabb467ac8db22db4a

File: pkg/render/intrusion_detection.go (hunk 1 above)
Comment (freecaykes) [reply pkg/render/intrusion_detection.go]: https://github.com/tigera/operator/pull/2074/commits/72b90570405d14a3140de0aabb467ac8db22db4a - set to ELASTIC_* in AD as well

File: pkg/render/intrusion_detection.go [hunk 3]
```diff
@@ -1440,6 +1441,19 @@ func (c *intrusionDetectionComponent) adDetectorAccessRole() *rbacv1.Role {
 					"update",
 				},
 			},
+			{
+				APIGroups: []string{
+					ADResourceGroup,
+				},
+				Resources: []string{
+					ADLogTypeMetaDataResourceName,
```
Comment (rene-dekker) [inline pkg/render/intrusion_detection.go:1449]: I think you could simply add "ADLogTypeMetaDataResourceName" to line 1435 and omit this block.

File: pkg/render/intrusion_detection_test.go [hunk 4]
```diff
@@ -361,31 +396,35 @@ var _ = Describe("Intrusion Detection rendering tests", func() {
...
+		assertEnvVarlistMatch(envs, expectedEnvs)
+
+		// expect AD PodTemplate EnvVars
+		expectedADEnvs := []expectedEnvVar{
+			{"ELASTIC_HOST", dns.GetServiceDNSNames(render.ESGatewayServiceName, render.ElasticsearchNamespace, cfg.ClusterDomain)[2], "", ""},
+			{"ELASTIC_PORT", strconv.Itoa(render.ElasticsearchDefaultPort), "", ""},
+			{"ELASTIC_CA", certificatemanagement.TrustedCertBundleMountPath, "", ""},
+			{"ELASTIC_USERNAME", "", render.ElasticsearchADJobUserSecret, "username"},
+			{"ELASTIC_PASSWORD", "", render.ElasticsearchADJobUserSecret, "password"},
+			{"MODEL_STORAGE_API_HOST", dns.GetServiceDNSNames(render.ADAPIObjectName, render.IntrusionDetectionNamespace, cfg.ClusterDomain)[2], "", ""},
```
Comment (rene-dekker) [inline pkg/render/intrusion_detection_test.go:414]: Maybe if you fully spell the value here, we never risk changes to   `dns.GetServiceDNSNames()` to break your feature.

File: pkg/render/intrusion_detection.go [hunk 5]
```diff
@@ -1479,6 +1493,39 @@ func (c *intrusionDetectionComponent) getBaseADDetectorsPodTemplate(podTemplateN
 		privileged = true
 	}
 
+	envVars := []corev1.EnvVar{
+		{
+			Name: "MODEL_STORAGE_API_HOST",
+			// static index 2 refres to - <svc_name>.<ns>.svc format
+			Value: dns.GetServiceDNSNames(ADAPIObjectName, IntrusionDetectionNamespace, c.cfg.ClusterDomain)[2],
```
Comment (rene-dekker) [inline pkg/render/intrusion_detection.go:1500]: I think it is better to just type it out and avoid  `dns.GetServiceDNSNames()`.

Comment (rene-dekker) [reply pkg/render/intrusion_detection.go:1500]: And perhaps store that value in a const.

**Review by rene-dekker — APPROVED** [review]
lgtm

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and delete the branch after I've merged it.

**Comment by marvin-tigera** [comment]
I'm sorry but I failed delete the branch after merging the pull request.

**Pull request merged**